pygmt/src/binstats.py

```diff
@@ -49,6 +48,7 @@ def binstats(
     ] = "number",
     quantile_value: float = 50,
     region: Sequence[float | str] | str | None = None,
+    registration: Literal["gridline", "pixel"] | bool = False,
```

Comment:
Docstring for registration should be updated to 1) add bool type hint, and 2) indicate that `registration=True` means pixel registration.

https://github.com/GenericMappingTools/pygmt/blob/25f265647de8097758ba7f42a141a50706590650/pygmt/helpers/decorators.py#L279-L283